### PR TITLE
Properly migrate clap to v4

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: cli/src/command/account_manager.rs [hunk 1]
```diff
@@ -46,11 +45,11 @@ pub enum AccountManagerCommand {
 
 #[derive(Debug, Clone, Args)]
 pub struct InitParameters {
```
Comment (kwek20) [inline cli/src/command/account_manager.rs:46]: does this need an `#[command(author, version)` as well?

Comment (thibault-martinez) [reply cli/src/command/account_manager.rs:46]: No

**Review by Thoralf-M — DISMISSED** [review]

**Review by kwek20 — DISMISSED** [review]

File: cli/Cargo.toml [hunk 2]
```diff
@@ -12,13 +12,13 @@ name = "wallet"
...
-tokio = { version = "1.25.0", default-features = false, features = [ "fs" ] }
\ No newline at end of file
+clap = { version = "4.1.14", features = [ "derive" ] }
+dialoguer = { version = "0.10.3", features = [ "history" ] }
+fern-logger = { version = "0.5.0" }
+log = { version = "0.4.17" }
+prefix-hex = { version = "0.7.0", features = [ "std" ] }
+serde_json = { version = "1.0.95" }
+thiserror = { version = "1.0.40" }
+tokio = { version = "1.27.0", features = [ "fs" ] }
```
Comment (Alex6323) [inline cli/Cargo.toml]: why do we explicitly set the patch versions in a binary?

Comment (thibault-martinez) [reply cli/Cargo.toml]: No big reason, convention and consistency

**Review by Alex6323 — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**